### PR TITLE
Added documentation for access restrictions

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
This PR adds the missing documentation for access restrictions.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

File: docs/config/access-restrictions.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+```yaml
+  regions:
+  - name: pangaea-north-1
+    zones:
+    - name: pangaea-north-1a
+    - name: pangaea-north-1b
+    - name: pangaea-north-1c
+    labels:
+      seed.gardener.cloud/eu-access: "true"
+```
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply docs/config/access-restrictions.md:21]: ✔️

File: docs/config/access-restrictions.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+Access restrictions are shown for clusters where the property `spec.seedSelector.matchLabels[key]` is set 
+and access restrictions options are shown the annotation `metadata.annotations[optionKey]` is set.
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply docs/config/access-restrictions.md:24]: ✔️

File: docs/README.md [hunk 3]
```diff
@@ -9,3 +13,6 @@
 ## Development
 
 TODO
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply docs/README.md:15]: ✔️

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**